### PR TITLE
Create Namespace: add static labels to namespace

### LINKED ISSUE #4922: Create Namespace: add static labels to namespace
**Summary**
Kubeapps allows to add namespaces to the cluster by the "create namespace" button. It should be possible to add static additional labels added to those newly created namespaces

**Background and rationale**
Currently the created namespaces by Kubeapps do not have any specific labels. But those labels would allow different Kubernetes components/operators/policy controllers to apply certain behavior. Like adding RoleBindings, NetworkPolicies, ... Currently that's only possible by either applying it to all newly created namespaces or figuring out through the initator with some rather complex logic.

**Description**
With these (static but configurable) labels it would even be possible to create a kubeapps with only "managed by kubeapps" namespaces. For that the namespaces could be filtered with a whitelisting label selector. In the other ticket I created earlier (#4921), I talked about a blacklist, to filter out certain namespaces. For the usecases I have in mind, it would work even better with such a whitelisting approach.

**Acceptance criteria**
Newly created namespaces are labeled with a preconfigured label.

### PR DESCRIPTION
**Pull request by smerschjohann**
This change adds the ability to add labels to namespaces during creation. These labels can be configured in the `config.json` of the dashboard.

### Description of the change

- Added labels to the grpc proto of the resources.proto
- passed these labels to the Kubernetes Manifest
- added `createNamespaceLabels` to the Dashboard configuration
- passed these values to the "createNamespace" method
- added the newly introduced config parameter to the Helm-Chart

### Benefits

See Issue

### Possible drawbacks

Current default behavior left unchanged.

### Applicable issues

- fixes #4922 

### Additional information

I seem to have a slightly different `buf` binary, as it produced a lot of changes in the generated code. I commited that as is, but as a seperate commit. I'm not sure how to solve that. 

I was also unable to get the newly introduced field into the openapi spec. What am I missing here?

### CODE REVIEW

**Comment by antgamdia** [comment]
Thanks for the contribution! We are currently in the middle of a release process (v2.4.6) and therefore we have a bunch of PRs pending review. We'll get to it as soon as possible! Anyway, it looks pretty good at a glance, thanks for the effort!

**Review by antgamdia — APPROVED** [review]
Great! Outstanding contribution: from a change in the proto file definition to the frontend code, besides changing the chart and other files. Well done!

Just for you to know, we are using [this project](https://github.com/bitnami-labs/readme-generator-for-helm) for autogenerating the readme.md, this way, each param you add to the values, will get auto-documented. No need to execute it for this PR, we can do it later.

I only have a minor comment (adding a test case), but otherwise, LGTM.

File: cmd/kubeapps-apis/plugins/resources/v1alpha1/namespaces.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func (s *Server) CheckNamespaceExists(ctx context.Context, r *v1alpha1.CheckName
 func (s *Server) CreateNamespace(ctx context.Context, r *v1alpha1.CreateNamespaceRequest) (*v1alpha1.CreateNamespaceResponse, error) {
 	namespace := r.GetContext().GetNamespace()
 	cluster := r.GetContext().GetCluster()
-	log.InfoS("+resources CreateNamespace ", "cluster", cluster, "namespace", namespace)
+	log.InfoS("+resources CreateNamespace", "cluster", cluster, "namespace", namespace, "labels", r.Labels)
```
[the remark this answers is not in the document]
Comment (smerschjohann) [reply cmd/kubeapps-apis/plugins/resources/v1alpha1/namespaces.go:51]: done

**Comment by smerschjohann** [comment]
@antgamdia ok, I added the test and rebased the PR to the latest changes

**Comment by castelblanque** [comment]
Many thanks @smerschjohann !!
We really appreciate your contribution.

**Pull request merged**